### PR TITLE
feat(ActionList): added stories and kitchensink for combinations

### LINKED ISSUE #1611: Add chromatic snapshot tests for ActionList
Current Dropdown snapshots don't take it's ActionList into consideration because its hidden in the initial snapshots. We should have tests of ActionList itself to avoid issues like #1610

### PR DESCRIPTION
**Pull request by harshalkaigaonkar**
closes #1611 

Changes Made: 

- Added 2 files, `ActionList.stories.tsx` containing story combinations for ActionList and `_KitcheSink.ActionList.stories.tsx` for it's Kitchensink dump.

Required TODO: @saurabhdaware can you give me some idea here about `argTypes` and `args`?  not sure how to show this on story docs.

### CODE REVIEW

File: packages/blade/src/components/ActionList/ActionList.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+ActionList.storyName = 'Default';
+ActionList.parameters = {
+  docs: {
+    description: {
+      story: 'Action List has Items inside as a list',
+    },
+  },
+};
+
+const ActionListWithLeadingComponentsExample: ComponentStory<typeof ActionListComponent> = () => {
```
Comment (saurabhdaware) [inline packages/blade/src/components/ActionList/ActionList.stories.tsx:126]: Lots of good examples in this file! This is perfect :D

**Comment by saurabhdaware** [comment]
I committed this change 25631c36e98920afa0cc42b2de5a5a45b3ced108 to fix lint issues and I moved ActionList story to `Dropdown/ActionList/Stories` since we already have some ActionList documentation there.

Overall looks good to me. I think it should be fine to not have controls on ActionList since its only meant for snapshots. We do have detailed documentation for it along with props in `Dropdown/` already.

**Comment by harshalkaigaonkar** [comment]
Hey @saurabhdaware , ig there's nothing to update here from my side, right?
is there any other issues to look up 👀 , wanted to explore the system.

**Review by saurabhdaware — APPROVED** [review]
Looks good to me. Yes you can take a look at issues and let me know if you find something interesting. 

Mostly the issues that are about adding props might be good to get started with. Some of these are marked as good first issues

**Comment by saurabhdaware** [comment]
@harshalkaigaonkar can you also fill the form mentioned in signup section of this page https://opensource.razorpay.com/hacktoberfest-2023?

**Pull request merged**